connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -4197,6 +4197,20 @@ object functions {
    */
   def split(str: Column, pattern: String): Column = Column.fn("split", str, lit(pattern))
 
+  /**
+   * Splits str around matches of the given pattern.
+   *
+   * @param str
+   *   a string expression to split
+   * @param pattern
+   *   a column of string representing a regular expression. The regex string should be a Java
+   *   regular expression.
+   *
+   * @group string_funcs
+   * @since 4.0.0
+   */
+  def split(str: Column, pattern: Column): Column = Column.fn("split", str, pattern)
```

Comment:
Let's add some new test cases for the connect module in `PlanGenerationTestSuite`.

Reply:
@CTCC1 also need to run `connect/ testOnly org. apache. spark. sql. connect. ProtoToParsedPlanTestSuite` to generate the golden files needed for reverse validation testing.

Reply:
thanks for the pointer, fixed